feat: implements the collection page

mamieorine opened the pull request:
Fixes #8842

Changes proposed in this pull request:

- Implements the web.dev collection page (/collection)

mamieorine commented:
@devnook

I have done implementing the collection page. Please review: https://deploy-preview-9253--web-dev-staging.netlify.app/collection/

Figma: https://www.figma.com/file/eXb7xAA7XROYOYNHkHcMJg/%5Bweb.dev%5D-Master?node-id=481%3A18737&t=pE3KlHQApVgmp2o6-0

I found that the content of a new design of the [collection](https://deploy-preview-9253--web-dev-staging.netlify.app/collection/) page has exactly the same as the  content of [/learn-core-web-vitals/](https://web.dev/learn-core-web-vitals/) page. I understand that a new collection page is the new redesign of [Learn Core Web Vitals](https://web.dev/learn-core-web-vitals/). So I decided to change the slug from _/learn-core-web-vitals/_ to _/collection_ and added the Topics section following the design on the Figma.

rachelandrew commented:
@mamieorine I don't understand the core web vitals change, that page should not change and should remain at the current URL. The collection page is a generic page for all collections, we just happened to use the core web vitals as a demo of content. Please revert this change to the core web vitals page.

mamieorine commented:
> @mamieorine I don't understand the core web vitals change, that page should not change and should remain at the current URL. The collection page is a generic page for all collections, we just happened to use the core web vitals as a demo of content. Please revert this change to the core web vitals page.

@rachelandrew I am fixing this now, I misunderstood the collection page and Ewa has already clarified this to me.  I will bring back the learn-core-web-vitals and reimplement the collection template.

mamieorine commented:
@devnook @rachelandrew  I have updated the collection template and brought back the learn-core-web-vitals page. Also, I renamed the template file from the path.njk to the collection.njk (commit: [9d369aa](https://github.com/GoogleChrome/web.dev/pull/9253/commits/9d369aacddadec900077dc6b84c3b41e07147798)) so you can see a lot of changed files.

mamieorine commented:
The layout of cards for each theme is flexible, so you can adjust the design, and position of a card by using CSS classes.

**Card layouts**
- no class specific for default card 
- `card-vertical` for displaying vertical card
- `card-horizontal` for displaying vertical card

**Theme colours**
- `bg-tertiary`
- `bg-quaternary`
- `bg-pink`
- `bg-dark`
- `bg-blue `

**Adjusts position**

The collection of cards offers a grid-based layout system, with rows and columns. You can adjust the card using `card-vertical` or `card-horizontal`

devnook commented on src/site/content/en/learn-core-web-vitals/index.njk:
```diff
@@ -1,6 +1,56 @@
 override:tags: []
 date: 2020-09-16
 pathName: learn-core-web-vitals
+cards:
```
This is an example usage of cards on a collection page. Please create a separate example page, somewhere under https://web.dev/handbook/#web.dev-markup, simiar to what @johnrich85  did in https://github.com/GoogleChrome/developer.chrome.com/tree/main/site/en/content-types and render these example cards there, with documentaion on how to use them.

mamieorine replied:
I created a document for the usage of cards on a collection page under [web.dev-markup](https://web.dev/handbook/#web.dev-markup).

- https://deploy-preview-9253--web-dev-staging.netlify.app/handbook/markup-cards-collection/ 

I also added the sample page of collection template in the document.

- https://deploy-preview-9253--web-dev-staging.netlify.app/example-collection/

devnook commented on src/site/content/en/learn-core-web-vitals/index.njk:
```diff
@@ -1,6 +1,56 @@
+    eyebrow:
+      icon: podcast
+      text: Podcast
+    cssClass: 'card-horizontal col-2 bg-tertiary'
```
We cannot expect authors to know internal css class names. Let's introduce 2-4 types of cards and assign a cardType to each. This should be documented for authors in the docs page.

mamieorine replied:
I added these YAML properties to allows the content creator to determine a card layout
- `cardLayout` 
- `theme`
- `column` 
- `row`

devnook commented on src/site/content/en/learn-core-web-vitals/index.njk:
```diff
@@ -1,6 +1,56 @@
+      icon: podcast
+      text: Podcast
+    cssClass: 'card-horizontal col-2 bg-tertiary'
+  - title: Meet the Chrome team
```
This type of card needs to link to /meet-the-team page, or generally have ability to add a target url to a card.

mamieorine replied:
I have added this feature.

devnook reviewed (changes requested):
The /core-web-vitals page still containd demo content from Figma. Please move demo content to a separate example page and leave core-web-vitals page as it was before.

Cards need to be linked to real pieces of content they highlight, reusing real content titles and graphics.

We need a batter way to stretch or highlight some of the cards, without a need to know css classes by the authors.

mamieorine commented:
@devnook I have fixed this PR as per your feedback. 

Here is the document for the usage of cards on a collection page
- https://deploy-preview-9253--web-dev-staging.netlify.app/handbook/markup-cards-collection/

mamieorine commented on src/site/content/en/learn-core-web-vitals/index.njk:
```diff
@@ -1,6 +1,56 @@
+    description: Praesent accumsan eros orci quis congue metus porta a sed dapibus magna nunc non finibus erat ullamcorper sit amet.
+    thumbnail: image/SZHNhsfjU9RbCestTGZU6N7JEWs1/MqeKXDtUPO7zszzpRXcF.png
+    eyebrow:
+      icon: featured
```
I have updated this issue.

devnook commented on src/site/content/en/example-collection/index.njk:
```diff
@@ -0,0 +1,59 @@
+---
```
Let's move this file to https://github.com/GoogleChrome/web.dev/blob/main/src/site/content/en/handbook/content-types/ folder

devnook commented on src/site/content/en/handbook/markup-cards-collection/index.md:
````diff
@@ -0,0 +1,163 @@
+{% raw %}
+```yaml
+---
+…
````
Let's remove three dots here, might be misleading

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+---
+layout: handbook
+title: Sample card layouts on the collection page
```
```suggestion
title: Add cards to the collection page
```

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+title: Sample card layouts on the collection page
+date: 2019-06-26
+description: |
+  A sample usage of cards on the collection page for web.dev
```
```suggestion
  How to add cards to the collection page for web.dev
```

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+  A sample usage of cards on the collection page for web.dev
+---
+
+The content and layout of cards are flexible, the content creator can change the content and adjust the design, and position of a card by using YAML properties.
```
```suggestion
The content and layout of cards are flexible, you can change the content, and adjust the design and position of a card by using YAML properties.
```

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+
+The content and layout of cards are flexible, the content creator can change the content and adjust the design, and position of a card by using YAML properties.
+
+The collection of cards offers a grid-based layout system, with rows and columns. The content creator can adjust the position card by specifying the card layout, vertical and horizontal, to avoid unwanted gaps. However, the rendering of a grid-based layout could create some gaps if we do not determine the column and row.
```
```suggestion
The collection of cards offers a grid-based layout system, with rows and columns. Adjust the position card by specifying the card layout, vertical and horizontal, to avoid unwanted gaps. 
```

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+
+See [the sample collection of cards](/handbook/content-types/example-collection/) post.
+
+YAML properties available for displying a layout of cards on collection template:
```
```suggestion
YAML properties available for displaying a layout of cards on a collection page:
```

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+
+### Adding default card
+
+This type of card the content creator probably needs to specific `title` and `description` YAML properties to display the content of the card. `url` property is needed if you want this card to be able to naviagate to somewhere.
```
```suggestion
This type of card requires a `title` and `description` YAML properties to display the content of the card. The `url` property is needed to link to another page.
```

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+
+### Adding vertical card
+
+The vertical card will represent a blog post, so the title, thumbnail, etc, will be automatically pulled from a real blog post, no need to specify the title, description, or thumbnail. However, the title, description, and thumbnail are able to be overridden.
+
+This type of card will consume two rows of the grid system. The rendering of a grid-based layout could create some gaps if we do not determine the column and row. For a horizontal card layout, you can determine row and column to specify the position using the YAML property `row` and `column`.
```
```suggestion
The vertical card represents a blog post, so the title, thumbnail, and description will be automatically pulled from the post specified in the `url`. However, the title, description, and thumbnail are able to be overridden by adding these properties.

This type of card will consume two rows of the grid system. The rendering of a grid-based layout could create some gaps if we do not determine the column and row. For a horizontal card layout, you can determine row and column to specify the position using the YAML property `row` and `column`.
```

rachelandrew commented on src/site/content/en/handbook/markup-cards-collection/index.md:
```diff
@@ -0,0 +1,91 @@
+
+### Adding horizontal card
+
+The horizontal card will represent a blog post, so the title, thumbnail, etc, will be automatically pulled from a real blog post, no need to specify the title, description, or thumbnail. However, the title, description, and thumbnail are able to be overridden.
```
```suggestion
The horizontal card represents a blog post, so the title, thumbnail, and description will be automatically pulled from  the post specified in the `url`. However, the title, description, and thumbnail are able to be overridden by adding these properties.
```

rachelandrew reviewed (approved):
This looks fine, I've just done a quick copyedit on the docs.

tunetheweb commented:
I think this PR has incorrectly added an extra tag to the main [Core Web Vitals](https://web.dev/vitals) page

![ED0F117B-AF67-4878-B5E0-09622B3BA070](https://user-images.githubusercontent.com/10931297/209005361-37529491-e31e-4aba-8315-6baf7e2fa4d4.jpeg)

And similarly in other pages: https://web.dev/vitals-business-impact/

Will open another issue too but wanted to comment here so those involved in this PR got visibility of it.
